storm-core/src/jvm/org/apache/storm/daemon/supervisor/SupervisorUtils.java

```diff
@@ -159,7 +162,7 @@ public static void addBlobReferences(Localizer localizer, String stormId, Map co
         }
     }
 
-    public static Set<String> readDownLoadedStormIds(Map conf) throws IOException {
+    public static Set<String> readDownLoadedStormIds(Map<String, Object> conf) throws IOException {
         Set<String> stormIds = new HashSet<>();
         String path = ConfigUtils.supervisorStormDistRoot(conf);
         Collection<String> rets = Utils.readDirContents(path);
```

Comment:
I agree that it could be improved. Let's create a separate jira issue to track it. I think it is out-of-scope here.